index.bs

```diff
@@ -314,12 +321,47 @@ See <a href="https://lists.w3.org/Archives/Public/public-speech-api/2012Sep/0072
+
+    When invoked, run these steps:
+    1. Let <var>promise</var> be <a>a new promise</a>.
+    1. Run the <a>on-device availability algorithm</a> with <var>lang</var> and <var>promise</var>. If it returns an exception, throw it and abort these steps.
+    1. Return <var>promise</var>.
+  </dd>
 
   <dt><dfn method for=SpeechRecognition>installOnDevice({{DOMString}} lang)</dfn> method</dt>
-  <dd>The installOnDevice method returns a Promise that resolves to a boolean indicating whether the installation of on-device speech recognition for a given BCP 47 language tag initiated successfully. [[!BCP47]]</dd>
+  <dd>
+    The {{SpeechRecognition/installOnDevice}} method returns a {{Promise}} that resolves to a {{boolean}} indicating whether the installation of on-device speech recognition for a given [[!BCP47]] language tag succeeded.
+
+    When invoked, run these steps:
+    1. If the [=current settings object=]'s [=relevant global object=]'s [=associated Document=] is NOT [=fully active=], throw an {{InvalidStateError}} and abort these steps.
+    1. If <var>lang</var> is not a valid [[!BCP47]] language tag, throw a {{SyntaxError}} and abort these steps.
+    1. If the on-device speech recognition language pack for <var>lang</var> is unsupported, return a resolved {{Promise}} with false and skip the rest of these steps.
+    1. Let <var>promise</var> be <a>a new promise</a>.
+    1. Initiate the download of the on-device speech recognition language for <var>lang</var>.
+        <p class=note>
+          Note: The user agent can prompt the user for explicit permission to download the on-device speech recognition language pack.
+        </p>
+    1. [=Queue a task=] on the [=relevant global object=]'s [=task queue=] to run the following step:
+        - If the download succeeds, resolve <var>promise</var> with <code>true</code>, otherwise resolve it with <code>false</code>.
+            <p class="note">
+              Note: The <code>false</code> resolution of the Promise does not indicate the specific cause of failure. User agents are encouraged to provide more detailed information about the failure in developer tools console messages. However, this detailed error information is not exposed to the script. This is non-normative.
+            </p>
+    1. Return <var>promise</var>.
+  </dd>
 
 </dl>
+<p>When the  <dfn>on-device availability algorithm</dfn> with <var>lang</var> and <var>promise</var> is invoked, the user agent MUST run the following steps:
+1. If the [=current settings object=]'s [=relevant global object=]'s [=associated Document=] is NOT [=fully active=], throw an {{InvalidStateError}} and abort these steps.
+1. If <var>lang</var> is not a valid [[!BCP47]] language tag, throw a {{SyntaxError}} and abort these steps.
+1. In parallel, determine the availability status for <var>lang</var>:
```

Comment:
Is there a particular reason we want to do this in parallel? When implementing, we likely have the value of this state synchronously available on the thread on which we need it, because any change to this state is caused by `installOnDevice`, and this changes the state on the thread itself.

Are we worried about UA specific bits of UI, or initial fetch of this value?